File: secretspec-swift/Examples/TypedAccess.swift
import Foundation
import SecretSpec

private struct AppSecrets: Decodable {
    let databaseURL: String

    private enum CodingKeys: String, CodingKey {
        case databaseURL = "DATABASE_URL"
    }
}

func typedAccess(resolved: Resolved) throws {
    let typed = try JSONDecoder().decode(
        AppSecrets.self,
        from: resolved.fieldsJSON()
    )
    print(typed.databaseURL)
}

File: secretspec-jvm/examples/typed_access/TypedAccess.java
import org.cachix.secretspec.SecretSpec;
import io.quicktype.Converter;

public class TypedAccess {

    public static void main() {
        try(var resolved = SecretSpec.builder().load()) {
            AppSecrets typed = Converter.fromJsonString(resolved.fieldsJson());
            System.out.println(typed.getDatabaseURL());
        }    
    }
}
